packages/rest/src/controllers/proofs/ProofController.ts

```diff
@@ -91,21 +150,29 @@ export class ProofController extends Controller {
    * to the connection with the specified connection id.
    *
    * @param proposal
-   * @returns ProofRecord
+   * @returns ProofExchangeRecordProps
    */
   @Post('/propose-proof')
-  @Example<ProofRecordProps>(ProofRecordExample)
+  @Example<ProofExchangeRecordProps>(ProofRecordExample)
   public async proposeProof(
     @Body() proposal: RequestProofProposalOptions,
     @Res() notFoundError: TsoaResponse<404, { reason: string }>,
     @Res() internalServerError: TsoaResponse<500, { message: string }>
   ) {
-    const { attributes, predicates, connectionId, ...proposalOptions } = proposal
+    const { attributes, predicates, connectionId, comment } = proposal
 
     try {
-      const presentationPreview = JsonTransformer.fromJSON({ attributes, predicates }, PresentationPreview)
-
-      const proof = await this.agent.proofs.proposeProof(connectionId, presentationPreview, proposalOptions)
+      const proof = await this.agent.proofs.proposeProof({
+        connectionId,
+        protocolVersion: 'v2',
```

Comment:
Here we are hard-coding v2 because this Agent setup supports only that protocol. I think `RequestProposalOptions` could be extended to accept other parameters, like `protocolVersion` and format (indy / anoncreds).